stepflow-rs/crates/stepflow-cli/src/test.rs

```diff
@@ -84,6 +84,75 @@ fn normalize_json_value(mut value: serde_json::Value) -> serde_json::Value {
     }
 }
 
+/// Check if actual FlowResult matches expected FlowResult using partial matching.
+/// Only fields present in `expected` are checked against `actual`.
+fn partial_match(expected: &FlowResult, actual: &FlowResult) -> bool {
+    match (expected, actual) {
+        (FlowResult::Success(exp), FlowResult::Success(act)) => {
+            partial_match_json(exp.as_ref(), act.as_ref())
+        }
+        (FlowResult::Failed(exp), FlowResult::Failed(act)) => {
+            // For errors, check code, message, and data if specified
+            if exp.code != act.code {
+                return false;
+            }
+            if exp.message != act.message {
+                return false;
+            }
+            // Only check data if expected has data specified
+            if let Some(exp_data) = &exp.data {
+                if let Some(act_data) = &act.data {
+                    if !partial_match_json(exp_data.as_ref(), act_data.as_ref()) {
+                        return false;
+                    }
+                } else {
+                    return false; // Expected data but actual has none
+                }
+            }
+            true
+        }
+        (FlowResult::Skipped { .. }, FlowResult::Skipped { .. }) => true,
+        _ => false, // Different variants don't match
+    }
+}
+
+/// Check if actual JSON value contains all fields from expected JSON value.
+/// Extra fields in actual are allowed.
+fn partial_match_json(expected: &serde_json::Value, actual: &serde_json::Value) -> bool {
+    use serde_json::Value;
+
+    match (expected, actual) {
```

Comment:
Similar to the above should we be propagating more information besides a boolen so we can construct more context on a `false` ?